scheduler/wrapper.go

```diff
@@ -155,10 +155,14 @@ func doStaticHostUpdate(d distro.Distro) ([]string, error) {
 			NeedsReprovision: provisionChange,
 			Provisioned:      provisioned,
 		}
-		if provisioned {
-			staticHost.Status = evergreen.HostRunning
+		if dbHost == nil || dbHost.Status == evergreen.HostTerminated {
+			if provisioned {
+				staticHost.Status = evergreen.HostRunning
+			} else {
+				staticHost.Status = evergreen.HostProvisioning
+			}
 		} else {
-			staticHost.Status = evergreen.HostProvisioning
```

Comment:
Were these hosts getting set to provisioning by this, and then to running by Evergreen?

Reply:
No, they were just being set to running because they had `provisioned = true`. The original statement didn't include the check for whether the host was in the db or its current status.